Tensile/Source/lib/include/Tensile/TensorDescriptor.hpp

```diff
@@ -342,11 +342,47 @@ namespace Tensile
 
     std::ostream& operator<<(std::ostream& stream, const TensorDescriptor& t);
 
+    template <typename T>
+    inline void WriteTensor1DHex(std::ostream& stream, T* data, TensorDescriptor const& desc)
+    {
```

Comment:
Might be able to write a smaller function to do the stream treatments:
template<typename T>
inline void WriteElement(std::ostream& stream, T* data, int index){  stream << data[index]; }

template<>
inline void WriteElement<float>(std::ostream& stream, float* data, int index){  stream << *reinterpret_cast<const uint32_t*>(&data[i]); }

...

And then plug this into the WriteTensor1DHex function to avoid specializing this bigger function and copying loops.